app/src/main/java/at/aau/edu/wizards/MainActivity.kt

```diff
@@ -43,15 +49,50 @@ class MainActivity : AppCompatActivity() {
     }
 
     private fun setupUi() {
+        val sharedPrefs: SharedPreferences = getSharedPreferences(getString(R.string.shared_storage), Context.MODE_PRIVATE)
+        if (usernameAlreadyExists(sharedPrefs)) {
+            binding.inputUsername.setText(sharedPrefs.getString(SHARED_PREFERENCE_USERNAME_KEY, ""))
+            setButtonsEnableStatus(true)
+        }
+        else {
+            setButtonsEnableStatus(false)
+        }
+
+        binding.inputUsername.addTextChangedListener(object : TextWatcher {
+            override fun beforeTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {}
```

Comment:
empty function declaration, add a nested comment explaining why it is empty. (this is a smell, i understand that you just want this to do nothing and need to implement it because class is abstract)